Change other to be a const SecurityOrigin& in SecurityOrigin

AlexsanderDamaceno opened the pull request:
#### f385189a193a735f1eba80a443512a5e49056fa2
<pre>
Change other to be a const SecurityOrigin&amp; in SecurityOrigin
<a href="https://bugs.webkit.org/show_bug.cgi?id=282589">https://bugs.webkit.org/show_bug.cgi?id=282589</a>

Reviewed by Chris Dumez.

Change other to be a const SecurityOrigin&amp; because function assumes it is non-null.

* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::equal const):
* Source/WebCore/page/SecurityOrigin.h:
* Source/WebCore/Modules/webdatabase/DatabaseManager.cpp:
(WebCore::DatabaseManager::fullPathForDatabase):
(WebCore::DatabaseManager::detailsForNameAndOrigin):
* Source/WebCore/dom/DocumentStorageAccess.cpp:
(WebCore::DocumentStorageAccess::hasStorageAccessQuickCheck):
(WebCore::DocumentStorageAccess::requestStorageAccessQuickCheck):
* Source/WebCore/loader/cache/MemoryCache.cpp:
(WebCore::MemoryCache::removeResourcesWithOrigin):
* Source/WebCore/page/SecurityPolicy.cpp:
(WebCore::SecurityPolicy::isAccessAllowed):
* Source/WebCore/storage/StorageEventDispatcher.cpp:
(WebCore::dispatchStorageEvents):
* Source/WebKitLegacy/mac/WebCoreSupport/WebSecurityOrigin.mm:
(-[WebSecurityOrigin isEqual:]):

Canonical link: <a href="https://commits.webkit.org/287111@main">https://commits.webkit.org/287111@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f2901a5dcfe22a25ee6ccf38d1f8001f98edf0ec

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78333 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/57379 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/31715 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/82994 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/29599 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/80466 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/66530 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5660 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61340 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19267 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/81400 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51353 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/68736 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41656 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48700 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/25026 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/27936 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69793 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/25393 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84359 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/5700 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/3861 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69566 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/5859 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67308 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68821 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17167 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12843 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/11158 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5647 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/8391 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5638 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9073 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/7425 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/SecurityOrigin.cpp:
```diff
@@ -588,22 +588,21 @@ Ref<SecurityOrigin> SecurityOrigin::create(WebCore::SecurityOriginData&& data, S
-bool SecurityOrigin::equal(const SecurityOrigin* other) const
+bool SecurityOrigin::equal(const SecurityOrigin& other) const
 {
     if (other == this)
```
`&other`

AlexsanderDamaceno replied:
@cdumez changed.

cdumez commented on Source/WebCore/page/SecurityPolicy.cpp:
```diff
@@ -198,7 +198,7 @@ bool SecurityPolicy::allowSubstituteDataAccessToLocal()
 bool SecurityPolicy::isAccessAllowed(const SecurityOrigin& activeOrigin, const SecurityOrigin& targetOrigin, const URL& targetURL, const OriginAccessPatterns& patterns)
 {
-    ASSERT(targetOrigin.equal(SecurityOrigin::create(targetURL).ptr()));
+    ASSERT(targetOrigin.equal(*SecurityOrigin::create(targetURL).ptr()));
```
`ASSERT(targetOrigin.equal(SecurityOrigin::create(targetURL)));` should work. If not,
`ASSERT(targetOrigin.equal(SecurityOrigin::create(targetURL).get()));`.

AlexsanderDamaceno replied:
@cdumez fixed

cdumez reviewed (approved):
LGTM with fix.

AlexsanderDamaceno committed:
Change other to be a const SecurityOrigin& in SecurityOrigin
https://bugs.webkit.org/show_bug.cgi?id=282589

Reviewed by Chris Dumez.

Change other to be a const SecurityOrigin& because function assumes it is non-null.

* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::equal const):
* Source/WebCore/page/SecurityOrigin.h:
* Source/WebCore/Modules/webdatabase/DatabaseManager.cpp:
(WebCore::DatabaseManager::fullPathForDatabase):
(WebCore::DatabaseManager::detailsForNameAndOrigin):
* Source/WebCore/dom/DocumentStorageAccess.cpp:
(WebCore::DocumentStorageAccess::hasStorageAccessQuickCheck):
(WebCore::DocumentStorageAccess::requestStorageAccessQuickCheck):
* Source/WebCore/loader/cache/MemoryCache.cpp:
(WebCore::MemoryCache::removeResourcesWithOrigin):
* Source/WebCore/page/SecurityPolicy.cpp:
(WebCore::SecurityPolicy::isAccessAllowed):
* Source/WebCore/storage/StorageEventDispatcher.cpp:
(WebCore::dispatchStorageEvents):
* Source/WebKitLegacy/mac/WebCoreSupport/WebSecurityOrigin.mm:
(-[WebSecurityOrigin isEqual:]):

Canonical link: https://commits.webkit.org/287111@main
